src/components/stacked-area-chart/StackedAreaChart.tsx

```diff
@@ -156,88 +161,143 @@ function maxLengthOfAllArrays(series: Series[]): number {
     return Math.max(...(series.map((serie) => serie.data.length)));
 }
 
-export const StackedAreaChart = ({title, subtitle, series, scale, scaleLabel}: StackedAreaChartProps) => {
-    stacker(series);
-    const colorGenerator = getColorGenerator();
-    let biggest = biggestNum(series);
-    let maxLen = maxLengthOfAllArrays(series);
-    let colors = series.map((datum) => datum.color ? datum.color : colorGenerator());
-    let polyPoints = polygonStackPoints(series);
-
-    return (
-        <div className="yarcl-chart stacked-area-chart">
-            {/*Title*/}
-            <div className="chart-title">
-                {title}
-            </div>
-            <div className="chart-subtitle">
-                {subtitle}
-            </div>
-            <svg viewBox={`0 0 ${chartX + marX + 100} ${chartY + marY + 50}`}>
-                <rect width="100%" height="100%" fill="white"/>
-                {/*Polygon*/}
-                {
-                    polyPoints.map((num, i ) =>
-                        <polygon
-                            key={i}
-                            points={num}
-                            fill={series[i].color || colors[i]}
-                            stroke="white"
-                            strokeWidth="1"
-                            opacity="1"
-                        />
-                    )   
-                }
-                    {
-                    scaleYAxis(scale, biggest)
-                    }
-                {/*Y axis*/}
-                <line 
-                    x1={marX}
-                    x2={marX} 
-                    y1={marY} 
-                    y2={marY + chartY}
-                    stroke="black"
-                    strokeWidth="4" 
-                />
-                {/*X axis*/}
-                <line 
-                    x1={marX - 2} 
-                    x2={marX + chartX - 0.5}
-                    y1={marY + chartY}
-                    y2={marY + chartY}
-                    stroke="black"
-                    strokeWidth="4" 
+export class StackedAreaChart extends React.Component<StackedAreaChartProps, StackedAreaChartState> {
+    state = {
+        hoverId: 0
+    };
+
+    copy(series: Series[]): Series[] {
+        return JSON.parse(JSON.stringify(series));
```

Comment:
ಠ_ಠ

Reply:
yep, there is some copying, as you said. ( ͡° ͜ʖ ͡°)